lib/src/list_extensions.dart

```diff
@@ -286,6 +287,29 @@ extension ListExtensions<E> on List<E> {
       yield slice(i, min(i + length, this.length));
     }
   }
+
+  /// Returns a new list with the elements of [other] are removed from `this`.
+  ///
+  /// It's aware about occurrence count and removes things only the amount
+  /// they are present in [other].
+  Iterable<E> subtract(List<E> other) sync* {
```

Comment:
I think we should document the assumptions we are making about `E`. I think mainly that it is is well behaved as a key in a `HashMap` (has a consistent `hashCode` and `equals`).

Reply:
Done I think